src/app/projects/components/project-settings/components/embeddings/embeddings.component.ts

```diff
@@ -104,25 +150,42 @@ export class EmbeddingsComponent implements OnInit, OnDestroy, OnChanges {
   addEmbedding() {
     if (!this.dataHandlerHelper.canCreateEmbedding(this.settingModals, this.embeddings, this.attributes)) return;
     const embeddingForm = this.settingModals.embedding.create.embeddingCreationFormGroup;
-    const embeddingHandle = embeddingForm.get("embeddingHandle").value;
     const attributeId = embeddingForm.get("targetAttribute").value;
-    const granularity = embeddingForm.get("granularity").value;
-
-    this.projectApolloService.createEmbedding(this.project.id, attributeId, embeddingHandle, granularity.substring(3)).pipe(first()).subscribe();
+    const platform = embeddingForm.get("platform").value;
+
+    const config = {
+      ...((platform == PlatformType.HUGGING_FACE || platform == PlatformType.PYTHON) && { model: embeddingForm.get("model").value }),
+      ...(platform == PlatformType.OPEN_AI && { model: embeddingForm.get("model").value, apiToken: embeddingForm.get("apiToken").value }),
+      ...(platform == PlatformType.COHERE && { apiToken: embeddingForm.get("apiToken").value }),
+      platform: platform,
+      termsText: this.gdprText.nativeElement.innerHTML,
+      termsAccepted: embeddingForm.get("termsAccepted").value,
+      embeddingType: embeddingForm.get("granularity").value.substring(3) === "TOKEN" ? EmbeddingType.ON_TOKEN : EmbeddingType.ON_ATTRIBUTE
+    };
```

Comment:
imo not really readable because of the initial condition logic. 

```suggestion
    const config = {    
      platform: platform,
      termsText: this.gdprText.nativeElement.innerHTML,
      termsAccepted: embeddingForm.get("termsAccepted").value,
      embeddingType: embeddingForm.get("granularity").value.substring(3) === "TOKEN" ? EmbeddingType.ON_TOKEN : EmbeddingType.ON_ATTRIBUTE
    };
        
if (platform == PlatformType.HUGGING_FACE || ....){
   config.model = embeddingForm.get("model").value   
}   
...    
    
```